packages/SwingSet/src/kernel/state/kernelKeeper.js

```diff
@@ -231,6 +232,15 @@ export default function makeKernelKeeper(kernelStorage, kernelSlog) {
 
   insistStorageAPI(kvStore);
 
+  // upgrade from pre-"vats.terminated" storage, or populate cache
+  let terminatedVats;
+  if (kvStore.has('vats.terminated')) {
+    terminatedVats = JSON.parse(getRequired('vats.terminated'));
+  } else {
+    terminatedVats = [];
+    kvStore.set('vats.terminated', JSON.stringify(terminatedVats));
+  }
```

Comment:
I've been considering abstracting this has+get/set pattern into something like
```js
  let terminatedVats = JSON.parse(getOrInit('vats.terminated', () => JSON.stringify([])));
```
No action necessary here, but I would like preliminary feedback on the concept.

Reply:
Yeah, it'd be nice to have some pattern for that. We almost always JSON-serialize entries like that, so I'd be tempted to bake the parse/stringify in, which would replace the thunk with a static `[]`, and combined that might make the usage short enough to not `eslint`-wrap into just as many lines as the original. We have a `getRequired()` helper already, this could sit next to it, and might even avoid an extra kvStore query (if it starts with `get` and has something meaningful to do if it gets back `undefined`).